File: DevExpress-Demo/CS/PropertyGridDemo.Wpf/ViewModels/TabbedViewModule/Model/SeriesOptions.cs
using DevExpress.Mvvm.Native;
using DevExpress.Mvvm.POCO;
using DevExpress.Xpf.Charts;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Media;

namespace PropertyGridDemo {
    [MetadataType(typeof(Metadata.SeriesOptionsMetadata))]
    public class SeriesOptions {
        public SeriesOptions() {
            Fill = FillType.CreateFillOptions(this);
            Border = BorderType.CreateBorderOptions(this);
            Options = ViewModelSource.Create<CommonSeriesOptions>();
            Mirror = ViewModelSource.Create<MirrorOptions>();
            Label = ViewModelSource.Create<LabelOptions>();
            InitializeOptions();
        }

        void InitializeOptions() {
            FillType = FillOptionsType.SolidFill;
            ((SolidFillOptions)Fill).Do(x => {
                x.Color = Colors.LightGray;
                x.Opacity = 0.5d;
            });
            BorderType = BorderOptionsType.Solid;
            ((SolidBorderOptions)Border).Do(x => {
                x.Color = (Color)ColorConverter.ConvertFromString("#FF093367");
                x.Opacity = 0.75;
                x.Thickness = 7d;
                x.DashStyle = DashStyles.Dash;
            });
            ShowLabel = true;
            Mirror.Show = true;
            Options.Model = new BorderlessSimpleBar2DModel();
        }

        [Category("FillnLine")]
        public virtual FillOptions Fill { get; set; }
        public virtual FillOptionsType FillType { get; set; }
        public void OnFillTypeChanged() {
            Fill = FillType.CreateFillOptions(this);
        }

        [Category("FillnLine")]
        public virtual BorderOptions Border { get; set; }
        public virtual BorderOptionsType BorderType { get; set; }
        public void OnBorderTypeChanged() {
            Border = BorderType.CreateBorderOptions(this);
        }
        [Category("Effects")]
        public virtual LabelOptions Label { get; set; }
        public virtual bool ShowLabel { get; set; }
        protected void OnShowLabelChanged() {
            Label = ShowLabel ? ViewModelSource.Create<VisibleLabelOptions>() : ViewModelSource.Create<LabelOptions>();
        }
        [Category("Effects")]
        public MirrorOptions Mirror { get; set; }
        [Category("SeriesOptions")]
        [DisplayName("Series Options")]
        public CommonSeriesOptions Options { get; set; }
    }
    [MetadataType(typeof(Metadata.CommonSeriesOptionsMetadata))]
    public class CommonSeriesOptions {
        public CommonSeriesOptions() {
            Model = Bar2DModel.GetPredefinedKinds().First();
        }
        public virtual object Model { get; set; }
    }
}

File: DevExpress-Demo/VB/PropertyGridDemo.Wpf/ViewModels/TabbedViewModule/Model/SeriesOptions.vb
Imports DevExpress.Mvvm.Native
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Xpf.Charts
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.Windows.Media

Namespace PropertyGridDemo

    <MetadataType(GetType(Metadata.SeriesOptionsMetadata))>
    Public Class SeriesOptions

        Public Sub New()
            Fill = FillType.CreateFillOptions(Me)
            Border = BorderType.CreateBorderOptions(Me)
            Options = ViewModelSource.Create(Of CommonSeriesOptions)()
            Mirror = ViewModelSource.Create(Of MirrorOptions)()
            Label = ViewModelSource.Create(Of LabelOptions)()
            InitializeOptions()
        End Sub

        Private Sub InitializeOptions()
            FillType = FillOptionsType.SolidFill
            CType(Fill, SolidFillOptions).[Do](Sub(x)
                x.Color = Colors.LightGray
                x.Opacity = 0.5R
            End Sub)
            BorderType = BorderOptionsType.Solid
            CType(Border, SolidBorderOptions).[Do](Sub(x)
                x.Color = CType(ColorConverter.ConvertFromString("#FF093367"), Color)
                x.Opacity = 0.75
                x.Thickness = 7R
                x.DashStyle = DashStyles.Dash
            End Sub)
            ShowLabel = True
            Mirror.Show = True
            Options.Model = New BorderlessSimpleBar2DModel()
        End Sub

        <Category("FillnLine")>
        Public Overridable Property Fill As FillOptions

        Public Overridable Property FillType As FillOptionsType

        Public Sub OnFillTypeChanged()
            Fill = FillType.CreateFillOptions(Me)
        End Sub

        <Category("FillnLine")>
        Public Overridable Property Border As BorderOptions

        Public Overridable Property BorderType As BorderOptionsType

        Public Sub OnBorderTypeChanged()
            Border = BorderType.CreateBorderOptions(Me)
        End Sub

        <Category("Effects")>
        Public Overridable Property Label As LabelOptions

        Public Overridable Property ShowLabel As Boolean

        Protected Sub OnShowLabelChanged()
            Label = If(ShowLabel, ViewModelSource.Create(Of VisibleLabelOptions)(), ViewModelSource.Create(Of LabelOptions)())
        End Sub

        <Category("Effects")>
        Public Property Mirror As MirrorOptions

        <Category("SeriesOptions")>
        <DisplayName("Series Options")>
        Public Property Options As CommonSeriesOptions
    End Class

    <MetadataType(GetType(Metadata.CommonSeriesOptionsMetadata))>
    Public Class CommonSeriesOptions

        Public Sub New()
            Model = Bar2DModel.GetPredefinedKinds().First()
        End Sub

        Public Overridable Property Model As Object
    End Class
End Namespace
